### PR TITLE
Add admin console endpoints for managing user tokens

### LINKED ISSUE #580: Add 'invalidate all tokens' for a user API endpoint
Add an endpoint which will invalidate all tokens for a user.

### LINKED ISSUE #760: Token management - admin console endpoints
An API should be provided for the admin console to allow administrative management of user's tokens, including diagnostic and revocation endpoints.

Related to #726

### PR DESCRIPTION
**Pull request by sbryzak**
Fixes #760 
Fixes #580 

Details to follow.

### CODE REVIEW

File: controller/user_blackbox_test.go [hunk 1]
```diff
@@ -283,6 +285,69 @@ func (s *UserControllerTestSuite) TestListUserResources() {
...
+	rt.AddScope("foo").AddScope("bar")
+	res := s.Graph.CreateResource(rt)
+
+	t1 := s.Graph.CreateToken(user, token2.TOKEN_TYPE_RPT)
+	priv := s.Graph.CreatePrivilegeCache(res, "foo", "bar")
+	t1.AddPrivilege(priv)
+
+	svc := testsupport.ServiceAsServiceAccountUser("User-Service", testsupport.TestAdminConsoleIdentity)
+	tokenManager, err := manager.NewTokenManager(s.Configuration)
+	require.Nil(s.T(), err)
```
Comment (alexeykazakov) [inline controller/user_blackbox_test.go]: require.NoError()

File: controller/user_blackbox_test.go [hunk 2]
```diff
@@ -283,6 +285,69 @@ func (s *UserControllerTestSuite) TestListUserResources() {
...
+
+func (s *UserControllerTestSuite) TestListUserTokensUnauthorized() {
+	user := s.Graph.CreateUser()
+
+	svc := testsupport.UnsecuredService("User-Service")
+	tokenManager, err := manager.NewTokenManager(s.Configuration)
+	require.Nil(s.T(), err)
+	ctrl := NewUserController(svc, s.Application, s.Configuration, tokenManager, nil)
+
+	test.ListTokensUserUnauthorized(s.T(), svc.Context, svc, ctrl, user.IdentityID().String())
```
Comment (alexeykazakov) [inline controller/user_blackbox_test.go]: Also test another service account.

File: design/token.go [hunk 3]
```diff
@@ -198,6 +198,21 @@ var _ = a.Resource("token", func() {
...
 	})
+
+	a.Action("RevokeAll", func() {
+		a.Routing(
+			a.GET("/revoke_all"),
+		)
+		a.Params(func() {
+			a.Param("identity_id", d.String, "Identifier of the identity for which all tokens will be revoked")
+			a.Required("identity_id")
+		})
```
Comment (alexeykazakov) [inline design/token.go]: Missed JWT security?

File: design/user.go [hunk 4]
```diff
@@ -41,6 +41,22 @@ var _ = a.Resource("user", func() {
 		a.Response(d.InternalServerError, JSONAPIErrors)
 		a.Response(d.Unauthorized, JSONAPIErrors)
 	})
+
+	a.Action("listTokens", func() {
```
Comment (alexeykazakov) [inline design/user.go]: Should be `/users/:identityID/tokens` instead.
Note, it's User**s** controller. Not User.

File: design/token.go [hunk 5]
```diff
@@ -198,6 +198,21 @@ var _ = a.Resource("token", func() {
 		a.Response(d.InternalServerError, JSONAPIErrors)
 		a.Response(d.Unauthorized, JSONAPIErrors)
 	})
+
+	a.Action("RevokeAll", func() {
+		a.Routing(
+			a.GET("/revoke_all"),
```
Comment (alexeykazakov) [inline design/token.go]: Maybe `DELETE /users/:identityID/tokens` instead?

File: controller/user_blackbox_test.go (hunk 1 above)
Comment (sbryzak) [reply controller/user_blackbox_test.go]: Fixed

File: controller/user_blackbox_test.go (hunk 2 above)
Comment (sbryzak) [reply controller/user_blackbox_test.go]: Done

File: design/token.go (hunk 3 above)
Comment (sbryzak) [reply design/token.go]: Good catch, I've added this.

File: design/user.go (hunk 4 above)
Comment (sbryzak) [reply design/user.go]: Ah yes, good point as the user controller is used for user-specific operations as opposed to user management.

File: design/token.go (hunk 5 above)
Comment (sbryzak) [reply design/token.go]: That makes sense, changed.

**Comment by alexeykazakov** [comment]
[test[

**Comment by sbryzak** [comment]
[test]

**Comment by sbryzak** [comment]
[test]

**Comment by riuvshin** [comment]
[test]

**Comment by riuvshin** [comment]
[test]

**Comment by riuvshin** [comment]
[test]

**Comment by riuvshin** [comment]
[test]

**Comment by riuvshin** [comment]
[test]

**Comment by riuvshin** [comment]
[test]

**Comment by riuvshin** [comment]
[test]

**Comment by sbryzak** [comment]
[test]

**Pull request merged**